### ISSUE TITLE
Grpc.Net.Client 2.52.0 missing port inside `:authority` header

### ISSUE THREAD

**Issue by AntonSmolkov** [grpc/grpc-dotnet#2099]
### What version of gRPC and what language are you using?

C#, Grpc.Net.Client 2.52.0

### What operating system (Linux, Windows,...) and version?

Linux

### What runtime / compiler are you using (e.g. .NET Core SDK version `dotnet --info`)

7.0.103

### What did you do?

I have multiple gRPC-services hosted inside one app on different ports, like this:

```C#
app.MapGrpcService<GrpcServiceOne>().RequireHost("*:5000");
app.MapGrpcService<GrpcServiceTwo>().RequireHost("*:5001");
app.MapGrpcService<GrpcServiceThree>().RequireHost("*:5002");
````

On the client-side, after updating  `Grpc.Net.Client` from version `2.51.0` to `2.52.0`, i started getting an error - 

> Bad gRPC response. HTTP status code - 404

 from  the services. 

At the same time Wombat(postman-like tool) works fine with them. Downgrading the library back to 2.48 also make the service work with the client again.

I analyzed  gRPC-requests of different versions and Wombat with Wireshark and found that  `2.52.0` doesn't have port defined inside `:authority` header. I think this is the reason of the issue.

Actual address of the service being tested is `http://localhost:5000`, resulting table:

|  Client |    :authority header | descripton |
|---|---|---|
| Grpc.Net.Client 2.51.0  | [::1]:5000  |  incorrent host, but fine because it has port |
| Grpc.Net.Client 2.52.0  | localhost  |  missing port |
| Wombat | localhost:5000 |  probably the most correct one  |

**Comment by JamesNK** [comment]
I'm taking a look at this. Could you show how you're creating the GrpcChannel? Are you using load balancing?

Also, what happens if you use [2.53.0-pre1](https://www.nuget.org/packages/Grpc.Net.Client/2.53.0-pre1)?

**Comment by JamesNK** [comment]
I think I have a fix: https://github.com/grpc/grpc-dotnet/pull/2111

I'm guessing you're using load balancing. If so, then this should be fixed in 2.54.0. Until then you'll need to stick with 2.51.0.

**Comment by AntonSmolkov** [comment]
> I'm taking a look at this. Could you show how you're creating the GrpcChannel? Are you using load balancing?
> 
> Also, what happens if you use [2.53.0-pre1](https://www.nuget.org/packages/Grpc.Net.Client/2.53.0-pre1)?

Hi! Yes, we are using client-side load-balancing with DnsResolverFactory. 

Here is the code (just in case):

```C#
            services.AddSingleton(provider =>
            {
                var credentials = CallCredentials.FromInterceptor(async (_, metadata) =>
                {
                    var securityService = provider.GetRequiredService<ISecurityService>();
                    var token = (await securityService.GetAccessToken()).AccessToken;
                    metadata.Add("Authorization", $"Bearer {token}");
                });

                var mvelEvaluationClientConfiguration = provider.GetRequiredService<IMvelEvaluationClientConfiguration>();
                var mvelEvaluationServiceUri = new Uri(mvelEvaluationClientConfiguration.MvelEvaluationServiceUrl);

                var channelOptions = new GrpcChannelOptions
                {
                    ServiceConfig = new ServiceConfig
                    {
                        LoadBalancingConfigs = { new RoundRobinConfig() }
                    },

                    // Can't use regular CompositeChannelCredentials because of the bug - https://github.com/grpc/grpc-dotnet/issues/1800, which is fixed in Grpc.Net.Client 2.52.0.
                    // Can't use Grpc.Net.Client 2.52.0 because of the bug - https://github.com/grpc/grpc-dotnet/issues/2099, which breaks interaction with external services.
                    // So use this workaround class until issues/2099 is resolved and fix is released.
                    Credentials = mvelEvaluationServiceUri.Scheme == "http"
                        ? new WorkaroundCompositeChannelCredentials(ChannelCredentials.Insecure, credentials)
                        : new WorkaroundCompositeChannelCredentials(ChannelCredentials.SecureSsl, credentials),
                    ServiceProvider = provider,
                    UnsafeUseInsecureChannelCallCredentials = true
                };

                // This "dns:///" enables client-side load balancing for k8s stateless services (multiple A-records, one for each pod).
                // https://docs.microsoft.com/en-us/aspnet/core/grpc/loadbalancing?view=aspnetcore-6.0
                var channel = GrpcChannel.ForAddress($"dns:///{mvelEvaluationServiceUri.Host}:{mvelEvaluationServiceUri.Port}", channelOptions);
                return new StatelessEvaluationService.StatelessEvaluationServiceClient(channel);
            });
```
